File: implementations/node/sdk/src/contracts/pipelines/pipeline-definition.ts
import type { AiSdkJsonValue } from "../../json/json-value.js";
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";
import type { AiSdkExecutionMode } from "./execution-mode.js";
import type { AiSdkPipelineStepDefinition } from "./pipeline-step-definition.js";

export interface AiSdkPipelineDefinition {
  readonly schemaVersion?: typeof AI_SDK_SCHEMA_VERSIONS.pipelineDefinition;
  readonly name: string;
  readonly version?: string;
  readonly executionLanguage?: string;
  readonly executionMode?: AiSdkExecutionMode;
  readonly steps?: readonly AiSdkPipelineStepDefinition[];
  readonly config?: Readonly<Record<string, AiSdkJsonValue>>;
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/pipelines/pipeline_definition.py
from __future__ import annotations

from dataclasses import dataclass, field

from ...json_types import AiSdkJsonValue
from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions
from .execution_mode import AiSdkExecutionMode
from .pipeline_step_definition import AiSdkPipelineStepDefinition


@dataclass(frozen=True)
class AiSdkPipelineDefinition(AiSdkWireModel):
    name: str
    schema_version: int = AiSdkSchemaVersions.PIPELINE_DEFINITION
    version: str | None = None
    execution_language: str | None = None
    execution_mode: AiSdkExecutionMode = AiSdkExecutionMode.SEQUENTIAL
    steps: tuple[AiSdkPipelineStepDefinition, ...] = ()
    config: dict[str, AiSdkJsonValue] = field(default_factory=dict, metadata={"json": True})
